### PR TITLE
[RFC] Possibility to hide fields in a list

### PR DESCRIPTION
**Pull request by fr05t1k**
What do think about this feature? Code is just an example (not a final version).

### CODE REVIEW

File: src/mui/list/Actions.js [hunk 1]
```diff
@@ -10,10 +11,23 @@ const cardActionStyle = {
...
+                     showFilter,
+                     refresh,
+                     handleFieldVisibility,
+                     fields,
+                     hiddenFields,
+}) => (
     <CardActions style={cardActionStyle}>
         {filters && React.cloneElement(filters, { resource, showFilter, displayedFilters, filterValues, context: 'button' }) }
         {hasCreate && <CreateButton basePath={basePath} />}
+        <HideFieldsButton fields={fields} handleFieldVisibility={handleFieldVisibility} hiddenFields={hiddenFields} />
```
Comment (djhi) [inline src/mui/list/Actions.js]: should be `onToggleFieldVisibility={onToggleFieldVisibility}`

File: src/mui/list/Datagrid.js [hunk 2]
```diff
@@ -76,13 +76,19 @@ class Datagrid extends Component {
         this.props.setSort(event.currentTarget.dataset.sort);
     }
 
+    filterFields = column => (
+        !column.props.source || this.props.hideFields.indexOf(column.props.source) === -1
```
Comment (djhi) [inline src/mui/list/Datagrid.js]: s/hideFields/hiddenFields/

File: src/mui/list/Datagrid.js [hunk 3]
```diff
@@ -76,13 +76,19 @@ class Datagrid extends Component {
         this.props.setSort(event.currentTarget.dataset.sort);
     }
 
+    filterFields = column => (
```
Comment (djhi) [inline src/mui/list/Datagrid.js]: rename `filterFields` to `isFieldVisible`

File: src/mui/list/HideFieldsButton.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    fields: [],
+};
+
+HideFieldsButton.propTypes = {
+    fields: React.PropTypes.arrayOf(React.PropTypes.shape({
+        title: React.PropTypes.string,
+        source: React.PropTypes.string,
+    })),
+    hiddenFields: React.PropTypes.arrayOf(React.PropTypes.string),
+    handleFieldVisibility: React.PropTypes.func.isRequired,
```
Comment (djhi) [inline src/mui/list/HideFieldsButton.js]: should be named `onToggleFieldVisibility`

File: src/mui/list/List.js [hunk 5]
```diff
@@ -183,6 +190,8 @@ export class List extends Component {
                         displayedFilters: this.state,
                         showFilter: this.showFilter,
                         refresh: this.refresh,
+                        handleFieldVisibility: this.toggleVisibilityField,
```
Comment (djhi) [inline src/mui/list/List.js]: s/handleFieldVisibility/onToggleFieldVisibility/

**Comment by fzaninotto** [comment]
This asks the question of the customizable datagrid. Because apart from adding/removing fields, one could want to reorder them, rename them, change the datatype, the number of results per page... So what's currently defined by configuration would have to move to runtime.

I'm not saying it's a bad idea, but I think it's a different type of datagrid, for a different usage. 

I advise you to publish it as a standalone package, which I'll be happy to link to in the documentation.

**Comment by fr05t1k** [comment]
Thanks for comments 😉

**Pull request closed without merging**